### PR TITLE
Release 1.1

### PR DESCRIPTION
**Pull request by warriorzz**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -21,4 +21,4 @@ jobs:
             ~/build
             ~/.gradle
       - name: Build with Gradle
-        run: chmod +x ./gradlew && ./gradlew compileKotlin
+        run: chmod +x ./gradlew && ./gradlew assemble test :backend:assemble :backend:test
```
Comment (NyCodeGHG) [inline .github/workflows/ci.yml:24]: remove the chmod calls and make gradlew executable

File: Dockerfile [hunk 2]
```diff
@@ -1,12 +1,12 @@
-FROM adoptopenjdk/openjdk15 as builder
+FROM adoptopenjdk/openjdk16 as builder
```
Comment (NyCodeGHG) [inline Dockerfile:1]: better use the openj9 jvm because of lower ram usage

File: Dockerfile [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
 
 COPY . .
 
 RUN chmod +x ./gradlew
 
 RUN ./gradlew --no-daemon installDist
 
-FROM adoptopenjdk/openjdk15
+FROM adoptopenjdk/openjdk16
```
Comment (NyCodeGHG) [inline Dockerfile:9]: also here

File: README.md [hunk 4]
```diff
@@ -1,4 +1,74 @@
...
+  dynamicbio:
+    env_file:
+      - .env # your .env file
+    image: warriorzz/dynamicbio # use warriorzz/dynamicbio:arm64-latest for arm64
+    restart: unless-stopped
+    container_name: dynamicbio
+    depends_on:
+      - mongo
+    environment:
+      - DATABASE_CONNECTION_STRING=mongodb://root:root@mongo/database?authSource=admin
```
Comment (NyCodeGHG) [inline README.md:20]:
```suggestion
      - DATABASE_CONNECTION_STRING=mongodb://root:root@mongo
```

File: README.md [hunk 5]
```diff
@@ -1,4 +1,74 @@
...
+    container_name: dynamicbiobackend
+    build:
+      context: .
+      dockerfile: backend.Dockerfile
+    networks:
+      - app
+    environment:
+      - DISCORD_TOKEN= # Discord token for your bot
+      - DATABASE_NAME=database
+      - DATABASE_CONNECTION_STRING=mongodb://root:root@mongo/database?authSource=admin
```
Comment (NyCodeGHG) [inline README.md:44]:
```suggestion
      - DATABASE_CONNECTION_STRING=mongodb://root:root@mongo
```

File: backend.Dockerfile [hunk 6]
```diff
@@ -0,0 +1,15 @@
+FROM adoptopenjdk/openjdk16 as builder
```
Comment (NyCodeGHG) [inline backend.Dockerfile:1]: also here

File: backend.Dockerfile [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+COPY . .
+
+RUN chmod +x ./gradlew
```
Comment (NyCodeGHG) [inline backend.Dockerfile:5]: also here the chmod thing

File: backend.Dockerfile [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+RUN ./gradlew --no-daemon :backend:installDist
+
+FROM adoptopenjdk/openjdk16
```
Comment (NyCodeGHG) [inline backend.Dockerfile:9]: open9j ftw

File: backend/build.gradle.kts [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+version = "1.0-SNAPSHOT"
+
+repositories {
+    mavenCentral()
+    maven("https://schlaubi.jfrog.io/artifactory/envconf/")
+    maven("https://oss.sonatype.org/content/repositories/snapshots")
+}
+
+dependencies {
+    implementation("dev.schlaubi", "envconf", "1.0")
```
Comment (NyCodeGHG) [inline backend/build.gradle.kts:17]:
```suggestion
    implementation("dev.schlaubi", "envconf", "1.1")
```

File: backend/src/main/kotlin/com/github/warriorzz/backend/Launcher.kt [hunk 10]
```diff
@@ -0,0 +1,13 @@
+package com.github.warriorzz.backend
+
+import com.github.warriorzz.backend.core.Database
+import com.github.warriorzz.backend.discord.DiscordBot
+import com.github.warriorzz.backend.server.startServer
+import mu.KotlinLogging
+
+suspend fun main() {
+    KotlinLogging.logger {}.info { "Starting service..." }
```
Comment (NyCodeGHG) [inline backend/src/main/kotlin/com/github/warriorzz/backend/Launcher.kt:9]:
```suggestion
    KotlinLogging.logger {}.info("Starting service...")
```

File: docker-compose-example.yml [hunk 11]
```diff
@@ -9,15 +9,35 @@ services:
     depends_on:
       - mongo
     environment:
-      - DATABASE_CONNECTION_STRING=
-      - DATABASE_NAME=
+      - DATABASE_CONNECTION_STRING=mongodb://root:root@mongo/database?authSource=admin
```
Comment (NyCodeGHG) [inline docker-compose-example.yml:12]:
```suggestion
      - DATABASE_CONNECTION_STRING=mongodb://root:root@mongo
```

File: docker-compose-example.yml [hunk 12]
```diff
@@ -9,15 +9,35 @@ services:
...
+    container_name: dynamicbiobackend
+    build:
+      context: .
+      dockerfile: backend.Dockerfile
+    networks:
+      - app
+    environment:
+      - DISCORD_TOKEN=
+      - DATABASE_NAME=database
+      - DATABASE_CONNECTION_STRING=mongodb://root:root@mongo/database?authSource=admin
```
Comment (NyCodeGHG) [inline docker-compose-example.yml:36]:
```suggestion
      - DATABASE_CONNECTION_STRING=mongodb://root:root@mongo/database
```

**Review by NyCodeGHG — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (warriorzz) [reply Dockerfile:1]: there is unfortunately no openj9 image for arm64.

**Comment by NyCodeGHG** [comment]
poggers

**Pull request merged**